projects/laji/src/app/shared-modules/select/select-subcategories/select-subcategories.component.ts

```diff
@@ -0,0 +1,570 @@
+    }
+
+    this.cd.detectChanges();
+    this.selectedChange.emit(this.selectedOptions);
+  }
+
+  emitSelectedOptions(cat)  {
+    this.selectedIdx[cat] = -1;
+    /*this.selectedOptions['GLOBAL'] = this.checkSubcategoriesExceptGlobalAreEquals(this.selectedOptions) ?
+    this.selectedOptions[this.subCategories[1]] : (this.selectedOptions['GLOBAL'] ? this.selectedOptions['GLOBAL'] : []);*/
+    this.checkSubcategoriesExceptGlobalAreEquals(this.selectedOptions);
+
+    this.initOptions(this.selectedOptions);
+    if (this.outputOnlyId) {
+      this.selectedChange.emit(this.selectedOptions);
+    } else {
+      this.selectedChange.emit(this.selectedOptions);
+    }
+  }
+
+  updateStatusSelectedOptionsGlobal(selectedOptionsGlobal, option, noGlobalOptionsSum, categoriesNoGlobal) {
+    if (Array.isArray(selectedOptionsGlobal) && selectedOptionsGlobal.length !== 0) {
+      selectedOptionsGlobal.push(
+        {...option, checkboxValue: noGlobalOptionsSum === categoriesNoGlobal.length ?
+          true : (noGlobalOptionsSum === 0 ? undefined : false)
+        }
+      );
+    } else {
+      selectedOptionsGlobal = [{...option, checkboxValue: noGlobalOptionsSum === categoriesNoGlobal.length ?
+        true : (noGlobalOptionsSum === 0 ? undefined : false)
+      }];
+    }
+
+    return selectedOptionsGlobal[0];
+  }
+
+  buildGlobalOptions(selectedOptions, categories, equals) {
+    const oldGlobalOptions = [];
+    (this.selectedOptions['GLOBAL'] || []).forEach(filter => {
+      if (filter.checkboxValue === true) {
+        oldGlobalOptions.push(filter.id);
+      }
+    });
+    this.selectedOptions['GLOBAL'] = [];
+    const tmpCat = [];
+
+    if (equals === false) {
+      for (let i = 0; i < categories.length; i++) {
+        selectedOptions[categories[i]].map(item => {
+          const index = tmpCat.findIndex(el => el.id === item);
+          if (index > -1) {
+            tmpCat[index]['count'] += 1;
+          } else {
+            tmpCat.push({'id': item, 'count': 1});
+          }
+        });
+     }
+
+     this.options.map(item => {
+       this.selectedOptions['GLOBAL'].push({...item, 'checkboxValue':
+       (tmpCat.filter(el => (el.id === item.id && el.count === this.subCategories.length - 1)).length > 0 ?
+       true : ((tmpCat.filter(el => (el.id === item.id && el.count > 0)).length > 0) ? false : undefined))});
+     });
+    } else if (equals === true) {
+      this.options.map(item => {
+        this.selectedOptions['GLOBAL'].push({...item, 'checkboxValue': true});
+      });
+    } else {
+      this.options.map(item => {
+        this.selectedOptions['GLOBAL'].push({...item, 'checkboxValue': undefined});
+      });
+    }
+  }
+
+  arrayDiff(a, b) {
+    a = a === undefined ? [] : a;
+    if (b.length >= a.length) {
+      return b.filter(({ id: id }) => !a.some(({ id: itemId }) => itemId === id));
+    } else {
+      return a.filter(({ id: id }) => !b.some(({ id: itemId }) => itemId === id));
+    }
+  }
```

Comment:
This method is sometimes returning sometimes a - b and sometimes b - a (depending which is bigger). Better algorithms for this can be found [here](https://stackoverflow.com/questions/1187518/how-to-get-the-difference-between-two-arrays-in-javascript)